[chongweiguan] iP

chongweiguan opened the pull request:
# DukeBot
> “Your mind is for having ideas, not holding them.” – David Allen ([an example](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

## Where do I begin?
All you need to do is,
1. download the JAR file from ([here](https://github.com/chongweiguan/ip/releases/tag/A-Jar)).
2. double-click it.
3. add your tasks
4. let it manage your tasks for you :smile:
And it is **FREE**

## Features
- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Managing events
- [ ] Managing 'do within a period' task

Commit:
Add Gradle support

chongweiguan committed:
testing

chongweiguan committed:
this is another test

chongweiguan committed:
Level 1. Greet, Echo, Exit

chongweiguan committed:
Level 2 complete

chongweiguan committed:
added Task.java and Level-3 completed

chongweiguan committed:
added Task.java and Level-4 completed

chongweiguan committed:
Level-5 completed

chongweiguan committed:
Level-6 completed

chongweiguan committed:
input.txt and expected.txt updated

chongweiguan committed:
added exceptions

chongweiguan committed:
Level-6 completed

chongweiguan committed:
A-Enums completed

chongweiguan committed:
Level 7 complete

chongweiguan committed:
Level 8 complete

chongweiguan committed:
Level 8 2nd commit

chongweiguan committed:
Merge branch 'branch-Level-8'

izzhafeez commented on src/main/java/Duke.java:
```diff
@@ -99,9 +103,12 @@ public static void main(String[] args) {
                         String to = command.substring(toStart);
-                        Event event = new Event(command.substring(0,fromStart-7), from, to);
+                        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+                        LocalDateTime formattedstartTime = LocalDateTime.parse(from, dateTimeFormatter);
+                        LocalDateTime formattedendTime= LocalDateTime.parse(to, dateTimeFormatter);
+                        Event event = new Event(command.substring(0,fromStart-7), formattedstartTime, formattedendTime);
```
Hi, I'm your commenter and I would like to confirm whether the camelCase for these variable declarations has been fixed? Otherwise, great job on the clean code!

izzhafeez reviewed (commented):
Looks good. Just some camelCase violations.

chongweiguan committed:
closed input

chongweiguan committed:
cleaned up a little

chongweiguan committed:
Classes committed

chongweiguan committed:
Merge branch 'branch-Level-8'

chongweiguan committed:
Revert "Classes committed"

This reverts commit 03dc3b0657c62410c186d1ea860e9e2f32e6aac2.

chongweiguan committed:
Delete Deadline.class

chongweiguan committed:
Delete DontKnowWhatThatMeansException.class

chongweiguan committed:
Delete Duke$EmptyDescriptionException.class

chongweiguan committed:
Delete Duke.class

chongweiguan committed:
Delete DukeExceptions.class

chongweiguan committed:
Delete EmptyDescriptionException.class

chongweiguan committed:
Delete Event.class

chongweiguan committed:
Delete Task.class

chongweiguan committed:
Delete Tasks.class

chongweiguan committed:
Delete Todo.class

chongweiguan committed:
testing

chongweiguan committed:
Added gradle

chongweiguan committed:
Complete A-MoreOOP

chongweiguan committed:
Complete A-Packages

chongweiguan committed:
Added JUnit Tests

chongweiguan committed:
Jar File Added

chongweiguan committed:
Complete A-JavaDoc

chongweiguan committed:
Completed coding standard

chongweiguan committed:
Completed find

afiqzu commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,88 @@
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
```
You might want to list out the specific imports instead of using a wildcard import

afiqzu commented on src/main/java/duke/UI.java:
```diff
@@ -28,6 +28,25 @@ public void showList(TasksList list) {
+            
+            String[] taskDescription = list.getTask(i).getDescription().split(" ");
+            for (String word : taskDescription) {
+                if(keyword.equals(word)) {
```
Minor coding standard violation but there should be a whitespace after 'if'

afiqzu reviewed (commented):
Code looks good overall bro 👍 Keep it up!

chongweiguan committed:
A-Checkstyle complete

chongweiguan committed:
Completed Level 10

chongweiguan committed:
Completed A-CheckStyle

chongweiguan committed:
Complete A-Assertions

chongweiguan committed:
Revert "Complete A-Assertions"

This reverts commit 8aaa678e04132bc346ee708e7f0d74d44ad7576e.

chongweiguan committed:
Complete A-Assertions

chongweiguan committed:
Complete A-CodeQuality

Code quality fixes on all java files, made method length shorter and added Javadoc comments

Good software engineering practices

Made method length shorter by abstracting out the function into separate methods, and added Javadoc comments for methods without JavaDoc comments

Abstracting out the functions into separate methods is preferable that helps with reading

chongweiguan committed:
Merge pull request #2 from chongweiguan/branch-Week-5

Merge A-Assertion

include assert statements in java files when required

chongweiguan committed:
Add JavaDoc and fix code quality

Documentation for class fields added for better readability

chongweiguan committed:
Merge pull request #3 from chongweiguan/master

A-CodeQuality

chongweiguan committed:
Update Storage.java

Added javaDoc comments

chongweiguan committed:
Merge pull request #4 from chongweiguan/branch-Week-5

A-CodeQuality

chongweiguan committed:
Update Duke.java

Added more JavaDoc for documentation and good software engineering practices

chongweiguan committed:
Complete DoWithinPeriodTasks

Add a feature that allows a user to add a task that needs to be done within a certain period

chongweiguan committed:
Fix code quality issues

Trim out trailing white spaces

chongweiguan committed:
Merge pull request #5 from chongweiguan/branch-A-CodeQuality

A-CodeQuality

chongweiguan committed:
Add greeting to Duke

chongweiguan committed:
Update Duke.java

chongweiguan committed:
Update DialogBox.java

Add JavaDoc description to ensure better readability and better documentation

chongweiguan committed:
Update MainWindow.java

Add JavaDoc for setDuke method in MainWindow.java

It allows for better readability and documentation is good software engineering practice

chongweiguan committed:
Add screenshot of UI

chongweiguan committed:
Update README.md

chongweiguan committed:
Fix jar file problems
